### ISSUE TITLE
BBB 2.3 Guest Lobby: Timeout issues

### ISSUE THREAD

**Issue by mk-hs** [bigbluebutton/bigbluebutton#12827]
**Describe the bug**
While using the guest lobby feature, users won't show up in the list of waiting users if they timed out and reloaded the page.

**To Reproduce**
1. Start a room with "Require Moderator approval before joining"
2. User tries to join. The user shows up in the guest list, but the moderator ignores it for the time being (scenario: the moderator is still talking to another user and only wants to allow in the waiting user when the other one has left). 
3. "no response from moderator" shows up after 8 minutes and 20 seconds, user disappears from the guest list.
4. User reloads the page, the lobby shows up again and everything appears normal to the user.

**Expected behavior**
The user shows up again on the guest list when the page is reloaded after the "no response" message. The moderator is then able to approve the user and let them in.

**Actual behavior**
Nothing happens for the moderator and everything appears normal to the user. The user even receives message updates, if the moderator pushes a message to the waiting room. However, the moderator see the user or can't let them in, so the user's waiting in vain.
The user's only showing up on the guest list again if they use the browser's back button and/or join again manually.

**BBB version:**
This has been present on at least 2.3.6 and still is on 2.3.8.
Greenlight 2.8.7 is being used as well, although control has been handed off to the BBB-server already when the guest lobby is active.

**Additional context**
The waiting room in itself is a bit of a weird feature, since the timeout is hardcoded to 8 minutes and 20 seconds in
`/usr/share/meteor/bundle/programs/server/assets/app/static/guest-wait/guest-wait.html` (access is being attempted every 5 seconds, 100 times).
It probably isn't designed to keep people waiting for hours, but the time limit seems arbitrary at best. It would be nice to have it easily configurable as well, instead of having to modify the html file on every update.
The timeout is probably pretty short so the server doesn't get hammered with requests if multiple waiting rooms are open, but if teachers are trying to use it for "face to face" consultations, only allowing in a few students at a time, the arbitrary timeout seems counterproductive. Nobody knows why there's an 8 minute timeout without looking at the code.

**Comment by PhMemmel** [comment]
Can reproduce this in 2.3.8.

Apparently the redirection is not working as intended: AFAIK there should be a redirect back to the greenlight/whateverfrontend page after a timeout of another 15s after seeing the "no answer" message. This doesn't happen. Both Firefox and Chrome aren't showing any errors in the console or anywhere else, so I think the redirect isn't being triggered at all.

Fixing this would kinda solve the problem of reloading the page which sends the user to the waiting room of infinity :-)
However, the fact that the user isn't being noticed that he isn't in the waiting room anymore if he reloads the site is pretty bad. This should be adressed I think.

**Comment by PhMemmel** [comment]
I can reproduce this behaviour on my 2.4-alpha2 machine as well. So the fix for this should also be ported to `develop`. Maybe also #12822 can be adressed when doing so.

**Comment by mk-hs** [comment]
Thanks for confirming my issue, and yeah it's completely broken in Alpha-2.

Redirecting to the frontend would help, but then there would be no explanation for why the user was redirected, pretty much defeating the purpose of the waiting room. The guest lobby message won't reach the frontend either.
If the user was waiting patiently, being returned to the frontend after a certain amount of time, he'd have to **know** that he timed out and must try again, because there's no feedback. He could also think that he has been rejected by the moderator and won't bother connecting again.

**Comment by PhMemmel** [comment]
I agree, instead of a redirect maybe two buttons/links like "join waiting room again" and "back to join page" would help there. Reloading the page should somehow be denied at the same time or at least made hard for the user to do so.

**Comment by mk-hs** [comment]
As a workaround, I've just set `const ATTEMPT_LIMIT` to a pretty high number, so the timeout doesn't trigger after 8 minutes. There's still no 'real' feedback, but the guest lobby message helps and the users don't end up in limbo.
I'd say 30 minutes would be an acceptable waiting time if there's no other reason why the timeout is so low.

**Comment by markginter31** [comment]
Latest update in 2.4 works without any workarounds.

**Comment by PhMemmel** [comment]
@markginter31 Just tried on my 2.4-beta1 dev environment: Bug's still there for me... ?

**Comment by Tainan404** [comment]
I was investigating the issue and found the bug, basically what happens is that the message `GuestsWaitingForApprovalEvtMsg` is sent only in user register on akka ([code](https://github.com/bigbluebutton/bigbluebutton/blob/df60295d228a6a7b3e6afce1fd3bd4b2df31e65b/akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/users/RegisterUserReqMsgHdlr.scala#L63)), and not in the polling on `guestWait` api, the request to `guestWait` is only used to sent the `GuestWaitingLeftMsg` if the user stops to send the request for X seconds ([code](https://github.com/bigbluebutton/bigbluebutton/blob/b0c66caef9e1c1f6d6b8267ffb65542309b176fb/bbb-common-web/src/main/java/org/bigbluebutton/api/MeetingService.java#L276)).

I've to solutions in mind:

1. register the user in guestWait endpoint 
2. redirect back to join endpoint


I prefer the second one because we don't merge responsibilities, but I'm open to suggestion

**Comment by pedrobmarin** [comment]
In our end we just do not timeout this users so we never had this problem with our users. Touching the register/join flow can be extremely complex and generate new problems.

**Comment by Commifreak** [comment]
We faced this issue as well. So, what exactly needs to be adapted to set this limit to an hour? I see changes needed in https://github.com/bigbluebutton/bigbluebutton/blob/342f6ad6c39e4dc3788c535b0a468cbcab374448/akka-bbb-apps/src/universal/conf/application.conf#L42 and https://github.com/bigbluebutton/bigbluebutton/blob/90953c9679abd95b2fc0c964de133d44a4cde509/bigbluebutton-html5/private/static/guest-wait/guest-wait.html#L265 ? Is that correct?

**Comment by mk-hs** [comment]
Just setting `const ATTEMPT_LIMIT` to 720 should be enough to increase the limit to an hour. 

720*5 seconds [from `const ATTEMPT_EVERY_MS = 5000`] = 60 minutes.
The mentioned PR would remove the limit altogether.

**Comment by Commifreak** [comment]
And this does not conflict with https://github.com/bigbluebutton/bigbluebutton/blob/342f6ad6c39e4dc3788c535b0a468cbcab374448/akka-bbb-apps/src/universal/conf/application.conf#L42 ?

What is https://github.com/bigbluebutton/bigbluebutton/blob/342f6ad6c39e4dc3788c535b0a468cbcab374448/akka-bbb-apps/src/universal/conf/application.conf#L42 thougt for?

See: https://github.com/bigbluebutton/bigbluebutton/pull/14131#issuecomment-1015809085

**Comment by sutidor** [comment]
Had the same issue (2.3), but we upgraded today and it seems to be resolved in latest 2.4. Thank you devs

**Comment by plastilincheg** [comment]
BBB 2.7 the same problem. 
The user is in the guest room - the moderator accepts him. But the user does not get into the conference, and continues to hang in the guest room.

**Comment by ffdixon** [comment]
Hi @plastilincheg, how many students do you have waiting in the guest room when the moderator accepts them all?

**Comment by plastilincheg** [comment]
Hi @ffdixon, I have one user in the queue.

**Comment by Tainan404** [comment]
I was not able to reproduce this issue in the latest version of BBB
